### DISCUSSION TITLE
How to access `template` function from `solid-js/web` on server side?

### DISCUSSION THREAD

**Discussion by adb-sh** [solidjs/solid-start#679] [General]
Hey :)
i wanted to create components without using jsx syntax for building component libraries. i would like to ship the dist folder without jsx when possible. but i wasn't able to find out how to access the `template` function on the server side unfortunately (`import { template } from "solid-js/web"` resolves to `undefined`). on the browser side its woking fine tho. i understand that there is no native dom available on the server, but is solid-start using the jsdom implementation or how is ssr achieved?

**Comment by ryansolid** [comment]
SSR is achieved with a different compilation that doesn't use template. It compiles to combining strings no DOM. We also generate different client code for hydration. Generally we recommend libraries ship the source under a `solid` export condition for this reason.

**Reply by adb-sh** [reply]
alright. thanks a lot for your insights :D
